File: 백준 1735.py
# -*- coding: utf-8 -*-
import sys
import math

def gcd(a, b):
    while b:
        a, b = b, a%b
    return a

a, b = map(int, input().split()) #a/b + c/d = over/below
c, d = map(int, input().split())

over = a*d + b*c
below = b*d

if over >= below:
    a, b = over, below
else:
    a, b = below, over
    
factor = gcd(a, b)
over //= factor
below //= factor

print(over, below)

File: 백준 1735.c
#include <stdio.h>

int gcd(int a, int b)
{
    if (b > a) {
        int temp = a;
        a = b;
        b = temp;
    }

    int r;
    while (b) {
        r = a % b;
        a = b;
        b = r;
    }

    return a;
}

int main()
{
    int a,b,c,d, over, below; // a/b + c/d = over/below
    scanf("%d %d", &a, &b);
    scanf("%d %d", &c, &d);

    over = a*d + b*c;
    below = b*d;

    int factor = gcd(over, below);
    over /= factor;
    below /= factor;

    printf("%d %d\n", over, below);
}